### ISSUE TITLE
Quarkus generates 'app.kubernetes.io/version' label as a part of the selectors for Deployment despite configuration

### ISSUE THREAD

**Issue by mgubaidullin** [quarkusio/quarkus#26522]
### Describe the bug

quarkus generates `app.kubernetes.io/version` label as a part of the selectors for **Deployment** despite configuration

```
quarkus.openshift.deployment-kind=Deployment
quarkus.openshift.add-version-to-label-selectors=false
```

For **DeploymentConfig** works as expected but for **Deployment** does not.

### Expected behavior

```
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: openshift-quickstart
```

### Actual behavior

```
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: openshift-quickstart
      app.kubernetes.io/version: "1234567890"
```

### How to Reproduce?

1. Create quarkus project
```
mvn io.quarkus.platform:quarkus-maven-plugin:2.10.1.Final:create \
   -DprojectGroupId=org.acme \
   -DprojectArtifactId=openshift-quickstart \
   -Dextensions="resteasy-reactive,openshift"
cd openshift-quickstart
```
2. Generate manifests
```
mvn clean package -DskipTests \
    -Dquarkus.container-image.build=true \
    -Dquarkus.container-image.insecure=true \
    -Dquarkus.container-image.builder=jib \
    -Dquarkus.container-image.group=demo \
    -Dquarkus.container-image.tag=1234567890 \
    -Dquarkus.kubernetes.deployment-target=openshift \
    -Dquarkus.openshift.deployment-kind=Deployment \
    -Dquarkus.openshift.add-version-to-label-selectors=false 
```
3. Check `target/kubernetes/openshift.yml`[](url)

### Output of `uname -a` or `ver`

Darwin Kernel Version 21.5.0

### Output of `java -version`

openjdk version "11.0.11"

### Quarkus version or git rev

2.10.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4

**Comment by geoand** [comment]
This looks like a bug in dekorate as we do [this](https://github.com/quarkusio/quarkus/blob/2.10.1.Final/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftProcessor.java#L277) in Quarkus.

@Sgitario would you like to investigate?

**Comment by Sgitario** [comment]
Fix in Dekorate: https://github.com/dekorateio/dekorate/pull/1010

**Comment by geoand** [comment]
Quarkus does not yet have a dekorate version that fixes this, correct?

**Comment by mgubaidullin** [comment]
Hello. Any news, when quarkus have a dekorate version that fix this?

**Comment by Sgitario** [comment]
> Hello. Any news, when quarkus have a dekorate version that fix this?

I didn't update this issue when bumping Dekorate, but it's now fixed. I've linked a pull request proving that this issue is gone.

**Comment by geoand** [comment]
@Sgitario which version fixes the issue?

**Comment by Sgitario** [comment]
> @Sgitario which version fixes the issue?

Dekorate 2.11.2

**Comment by geoand** [comment]
I meant Quarkus :)

When we close issues, we do try to set the milestones to help future readers (and ourselves)

**Comment by Sgitario** [comment]
ah ok, the fix was included in `2.12.1.Final`. I've already updated the issue with the milestone.

**Comment by geoand** [comment]
Excellent, thanks
